ERCS/erc-7567.md

```diff
@@ -0,0 +1,155 @@
+The contract provides functions for retrieving information about utilities and their association with NFTs:
+- `getUtilitiesOf(uint256 _tokenID)`: It retrieves the list of utility IDs associated with a specific NFT's tokenID.
+- `allUtilitiesList()`: This function retrieves the complete list of available utilities, including their IDs, names, issuers, and transferability status.
+- `allUtilityIndexList()`: This function returns the list of IDs of all available utilities.
+- `utilityMap(uint256 id)`: Returns the details of a specific utility based on its ID.
+
+**Interface**
+```solidity
+// SPDX-License-Identifier: CC0
+
+pragma solidity ^0.8.20;
+
+/// @title Multiutility Contract Interface
+/// @dev Interface for the Multiutility contract
+/// @author []
+
+interface Multiutility {
+struct Utility {
+​ uint256 id;
+​ string name;
+​ address issuer;
+​ bool transferable;
+}
+
+/**
+* @dev Mint a new utility NFT with the specified utility indices.
+* @param utilityIndex Array of utility indices to be associated with the NFT.
+*/
+
+function mintUtilityNFT(uint256[] memory utilityIndex) external;
+/**  
+* @dev Mint a new utility NFT with all available utilities.
+*/
+function mintFullNFT() external;
+
+/**
+* @dev Transfer utilities from one token to another, and transfer the new token to the receiver.
+* @param tokenId The ID of the token containing the utilities to be transferred.
+* @param receiver The address to receive the new token with transferred utilities.  
+*/
+
+function transferUtilities(uint256 tokenId, address receiver) external;
+
+/**
+* @dev Add a new utility to the contract.
+* @param utility The utility struct to be added.
+*/ 
+
+function addUtilities(Utility memory utility) external;
+
+/**
+* @dev Mint a new utility to an existing NFT.
+* @param tokenId The ID of the existing NFT.
+* @param utilityIndex The index of the utility to be minted to the NFT.
+*/
+
+function mintNewUtilityToNFT(uint256 tokenId, uint256 utilityIndex) external;
+
+/**
+* @dev Get the utilities associated with a specific token.
+* @param \_tokenID The ID of the token to query for utilities. 
+* @return An array of utility indices associated with the token.
+*/
+
+function getUtilitiesOf(uint256 \_tokenID) external view returns (uint256[] memory);
+
+/**
+* @dev Get a list of all utilities defined in the contract.
+* @return An array of utility structs representing all utilities.
+*/
+
+function allUtilitiesList() external view returns (Utility[] memory);
+
+/**
+* @dev Get a list of all utility indices defined in the contract.
+* @return An array of utility indices.
+*/
+
+function allUtilityIndexList() external view returns (uint256[] memory);
+
+/**
+* @dev Get information about a specific utility based on its ID.
+* @param id The ID of the utility to query.
+* @return A utility struct representing the specified utility.
+*/
+
+function utilityMap(uint256 id) external view returns (Utility memory);
+}
+```
+
+## Rationale
```

Comment:
A lot of your rationale below justifies the existence of the EIP as a whole. The Rationale section should explain technical choices made within the EIP itself.

> Motivation: "Why do we need a shed?"
> Rationale: "Why did we paint the shed red?"